### ISSUE TITLE
[jwt-play] Having two header name configuration

### ISSUE THREAD

**Issue by Isammoc** [jwt-scala/jwt-scala#61]
(Feature request)

With Play 2.6 and default back-end (akka-http), the `Authorization` header is currently not rendered in response. IMHO, this is a bug (`Authorization` header should be a custom header in response).

`play.http.session.jwtName` should be separated in two different configuration : 

 * one for request header `play.http.session.jwt.requestName` with current default: `Authorization`
 * one for response header `play.http.session.jwt.responseName` with another default: `Set-Authorization` (As for the pair `Cookie` / `Set-Cookie`)

This feature request follow the discussion in playframework/playframework#7549

If this feature is agreed, I can submit a PR.

**Comment by pauldijou** [comment]
Another user already reported the problem : https://github.com/pauldijou/jwt-scala/issues/50

I'm not super kind to the idea of introducing a breaking change on the lib for a problem which is more about Akka HTTP than JWT Scala.  Keep me updated about the Akka / Play framework team position, I would prefer for everyone to be able to use whatever HTTP headers they want rather than introducing this workaround.

**Comment by Isammoc** [comment]
The default behaviour is already broken for now with Play 2.6.

I strongly understand your position about the breaking change.
What about adding just one another configuration key for the server part?

Something like that :

```INI
# Header name for the request header where the library can find the JWT token used for the session
# Default : play.http.session.jwtName = "Authorization"
play.http.session.jwtName = "Authorization"

# Header name for the response header name where the library put the JWT token
# Default to the same value as play.http.session.jwtName. 
play.http.session.jwtResponseName = null
```

And in the source:

```scala
 val REQUEST_HEADER_NAME: String = getConfigString("play.http.session.jwtName").getOrElse("Authorization")
 val RESPONSE_HEADER_NAME: String = getConfigString("play.http.session.jwtResponseName").getOrElse(getConfigString("play.http.session.jwtName").getOrElse("Authorization"))
```

In this way, there are no breaking change, and the new feature is available for the user (like me) of the library who want different headers.

**Comment by Isammoc** [comment]
About the issue #50: Sorry I didn't see it, I only looked for open issues.

About your last comment in #50: The [RFC7235](https://tools.ietf.org/html/rfc7235#section-4.2) defines the `Authorization` as the valid header to authenticate the client. But nothing (as far as I seen) in RFCs set the `Authorization` header as invalid for response.

In another hand, we have an example with `Cookie` (request) and `Set-Cookie` (response) from [RFC6265](https://tools.ietf.org/html/rfc6265#section-4.1) and I guess there are other examples for this mechanism.

**Comment by pauldijou** [comment]
Ok, we can do that. A test with the PR would be nice but don't spend too much time on it, I can add it quickly after merging.

```scala
 val REQUEST_HEADER_NAME: String = getConfigString("play.http.session.jwtName").getOrElse("Authorization")
 val RESPONSE_HEADER_NAME: String = getConfigString("play.http.session.jwtResponseName").orElse(getConfigString("play.http.session.jwtName").getOrElse(REQUEST_HEADER_NAME))
```

**Comment by pauldijou** [comment]
Released in 0.14.0
